src/gui/attributetable/qgsattributetablemodel.cpp

```diff
@@ -74,7 +74,12 @@ QgsAttributeTableModel::QgsAttributeTableModel( QgsVectorLayerCache *layerCache,
 
   connect( mLayer, &QgsVectorLayer::editCommandStarted, this, &QgsAttributeTableModel::bulkEditCommandStarted );
   connect( mLayer, &QgsVectorLayer::beforeRollBack, this,  &QgsAttributeTableModel::bulkEditCommandStarted );
-  connect( mLayer, &QgsVectorLayer::afterRollBack, this, &QgsAttributeTableModel::bulkEditCommandEnded );
+  connect( mLayer, &QgsVectorLayer::afterRollBack, this, [ = ]
+  {
+    mIsRollingBack = true;
```

Comment:
Should this be set to true when the ::beforeRollBack signal is received ? Would it make more sense ?

Reply:
I think that it's fine here, it's only used for this case and it is immediately clear that it is set right before and unset right after the call to bulkEditCommandEnded();

Reply:
> > 
> 
> I'll have a look, but if I've not already wrote a test for it there are reasons:
> 
>     1. could quite of an effort because it involves a few classes and requires a transactional DB (not sure if OGR-based spatialite or GPKG would be sufficient to trigger the issue)
> 
>     2. it's GUI and tests for GUI are not required
> 
> 
> I agree a test never hurts but there are probably more urgent matters to address given the limited time and resources that we have.

Personally I think simplifying the writing and running of tests is the most urgent matter, in QGIS.
PostgreSQL is transactional, having a way for tests to ONLY RUN IF an accessible PostgreSQL cluster is available (time to make it a FIXTURE ?) would be great.
Is there currently NO test at all for QGIS GUIs ? I see a tests/src/gui/ directory.

Reply:
I do understand it is out-of-scope with the specific work of fixing *this* bug but maybe spending some time to do some research for tools helping with GUI testing will pay in the long run. I've found (but not tested) this project which looks interesting: http://pedromateo.github.io/openhmitester/

Reply:
I've added the link to this new wiki page I created:
https://github.com/qgis/QGIS-Enhancement-Proposals/wiki/Testing-frameworks

Reply:
You have to create an issue: https://github.com/qgis/QGIS-Enhancement-Proposals/issues
Anyway, QT already has a decent GUI testing framework.

Reply:
Yeah, I was going to do that but I realized I don't know the *existing* framework good enough to tell if we need something better. In any case a GUI based test *generator* like the one linked above would seem to make it very easy to create test for user reported bugs like this one, where the user uploaded a screencast of what is done to trigger a crash. Will probably create a lot of bloat but could be an effective way to guard after some fixes (maybe a separate repository for those kind of tests would be advisible)